### PR TITLE
Initial derivation remote cache related metrics: check, download and upload time.

### PR DESCRIPTION
**Pull request by restingbull**
* Introduces building block datum, LocalActions, to make it easier to reason about actions.
* Introduces RemoteCacheMetrics as basic reporting datum for cache performance.

### CODE REVIEW

**Comment by saraadams** [comment]
@restingbull I haven't had a look at this, as I wanted to give you time to address Louis' comments first. Let me know if you are waiting for my review first.

**Comment by restingbull** [comment]
Been a bit busy to address all the comments. Will get to it this week.

**Comment by afrueda97** [comment]
I've added the suggestions to the pull request @lrgoldstein , what do you think?

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/LocalActions.java
Comment (lrgoldstein) [inline analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/LocalActions.java:77]: Other providers just return `null`. Suggest being consistent.

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/LocalActionsDataProvider.java
Comment (lrgoldstein) [inline analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/LocalActionsDataProvider.java:39]: Nit: too much space here.

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetricsDataProvider.java
Comment (restingbull) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetricsDataProvider.java:50]: co·a·lesce
/ˌkōəˈles/
verb
come together to form one mass or whole.

I suppose we could use aggregate or summarize?

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetricsDataProvider.java
Comment (restingbull) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetricsDataProvider.java:68]: Future work, really. Was intending to add some actual useful data to the metrics.

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java
Comment (restingbull) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java:33]: That's a bit of an unexpected api.

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java
Comment (restingbull) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java]: Added more description.

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java
Comment (restingbull) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java:60]: Tests.

**Comment by restingbull** [comment]
> > I've added the suggestions to the pull request @lrgoldstein , what do you think?
> 
> I've gone through and resolved the items that you addressed and added a couple... there were also a handful from before that weren't addressed (be sure to "Load" the "hidden conversations" in the middle).
> 
> Edit: Also, you have to [sign all your commits](https://github.com/EngFlow/bazel_invocation_analyzer/blob/main/CONTRIBUTING.md#process).

You have to do a bit more that just sign them. I have `commit.gpgsign true` (hence I have Screen Shot 2023-03-01 at 8 29 34 PM) -- which means I can prove I made the commit...

You have to a more explicit `--sign-off`, which means you stand by it (I guess?). Thanks, git, for not making this confusing.

File: analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java
Comment (lrgoldstein) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/RemoteCacheMetrics.java]: You would expect it to have an Empty Reason if it's not Empty? It's typically determined dynamically, as you could run into a number of different problems while producing a datum depending on the complexity of the calculation.

**Review by lrgoldstein — APPROVED** [review]

**Pull request merged**
